decidim-core/app/packs/src/decidim/direct_uploads/upload_field.js

```diff
@@ -1,117 +1,135 @@
 import UploadModal from "src/decidim/direct_uploads/upload_modal";
-import { truncateFilename, createHiddenInput } from "src/decidim/direct_uploads/upload_utility";
+import { truncateFilename } from "src/decidim/direct_uploads/upload_utility";
 
-const loadAttachments = (modal) => {
-  Array.from(modal.activeAttachments.children).forEach((child) => {
-    modal.createUploadItem(child.dataset.filename, child.dataset.title, "validated");
-  })
+const updateModalTitle = (modal) => {
+  if (modal.uploadItems.children.length === 0) {
+    modal.modalTitle.innerHTML = modal.modalTitle.dataset.addlabel;
+  } else {
+    modal.modalTitle.innerHTML = modal.modalTitle.dataset.editlabel;
+  }
+  modal.updateDropZone();
 }
 
-const addInputEventListener = (modal) => {
-  modal.input.addEventListener("change", (event) => {
-    event.preventDefault();
-    const files = event.target.files;
-    Array.from(files).forEach((file) => modal.uploadFile(file));
-  })
-}
+const updateActiveUploads = (modal) => {
+  // remove the default image block, if exists
+  const defaultFile = document.getElementById(`default-active-${modal.modal.id}`)
+  if (defaultFile) {
+    defaultFile.remove()
+  }
+
+  const files = document.querySelector(`[data-active-uploads=${modal.modal.id}]`)
+  const previousId = Array.from(files.querySelectorAll("[type=hidden][id]"))
+  const isMultiple = modal.options.multiple
+
+  // fastest way to clean children nodes
+  files.textContent = ""
+
+  // divide the items between those will gonna be removed, and added
+  const [removeFiles, addFiles] = [modal.items.filter(({ removable }) => removable), modal.items.filter(({ removable }) => !removable)]
+
+  addFiles.forEach((file, ix) => {
+    let title = truncateFilename(file.name, 19)
+
+    let hidden = ""
+    if (file.hiddenField) {
+      // if there is hiddenField, this file is new
+      // eslint-disable-next-line no-ternary
+      const fileField = isMultiple
+        ? `${modal.options.resourceName}[${modal.options.addAttribute}][${ix}][file]`
+        : `${modal.options.resourceName}[${modal.options.addAttribute}]`
 
-const addButtonEventListener = (modal) => {
-  modal.button.addEventListener("click", (event) => {
-    event.preventDefault();
-    Array.from(modal.trashCan.children).forEach((item) => {
-      modal.uploadItems.append(item);
-    })
-    if (modal.uploadItems.children.length === 0) {
-      modal.modalTitle.innerHTML = modal.modalTitle.dataset.addlabel;
+      hidden = `<input type="hidden" name="${fileField}" value="${file.hiddenField}" />`
     } else {
-      modal.modalTitle.innerHTML = modal.modalTitle.dataset.editlabel;
+      // otherwise, we keep the attachmentId
+      // eslint-disable-next-line no-ternary
```

Comment:
We should probably change this eslint rule, as we have more than 10 usage of ternarys in javascript: 

```shell
$ rg "no-ternary" . |grep disable | wc -l
10
```

I understand that the reasoning is that some more junior developers may have trouble understanding a ternary, but as its an operator that's used in ruby too, then I don't see any issue.

*In Ruby we're using it at least 10 times more :

```shell
$ rg "= .* \? .* : .*" decidim-*/app/**/*.rb |wc -l
111
```

cc @decidim/maintainers

Reply:
Reported on #11730 by @ferblape